src/cpu/cpu.rs

```diff
@@ -23,6 +23,71 @@ pub struct Cpu {
     interconnect: interconnect::Interconnect
 }
 
+impl Display for Cpu {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        
+        let reg_name = [
+        "r0", "at", "v0", "v1", "a0", "a1", "a2", "a3",
+        "t0", "t1", "t2", "t3", "t4", "t5", "t6", "t7",
+        "s0", "s1", "s2", "s3", "s4", "s5", "s6", "s7",
+        "t8", "t9", "k0", "k1", "gp", "sp", "s8", "ra",
+        ];
+        
+        for reg_num in 0..32 {
+            try!(
+                writeln!(f, 
+                    "{reg_name}/reg{num:02}: {value:#018X}",
+                    num = reg_num,
+                    reg_name = reg_name[reg_num],
+                    value = self.reg_gpr[reg_num],
+                )
+            );
+        }
+        
+        try!(
+            writeln!(f,"")
+        );
+            
+        for reg_num in 0..32 {
+            try!(
+                writeln!(f, 
+                    "fp_reg{num:02}: {value}",
+                    num = reg_num,
+                    value = self.reg_fpr[reg_num],
+                )
+            );
+        }
+        
+        try!(
+            writeln!(f,"")
+        );
+        
+        try!(    
+            writeln!(f, 
+                "\
+                reg_pc: {}\n\
+                reg_hi: {}\n\
+                reg_lo: {}\n\
+                reg_llbit: {}\n\
+                reg_fcr00: {}\n\
```

Comment:
Alignment. I can either add a space or leave it unaligned whatever works better for you.